markdown/renderer.go

```diff
@@ -23,22 +23,35 @@ var (
 	codeBlockChars          = []byte("```")
 	tableHeaderColChar      = []byte{'-'}
 	tableHeaderAlignColChar = []byte{':'}
+	heading1UnderlineChar   = []byte{'='}
+	heading2UnderlineChar   = []byte{'-'}
 )
 
 // Ensure compatibility with Goldmark parser.
 var _ renderer.Renderer = &Renderer{}
 
 // Renderer allows to render markdown AST into markdown bytes in consistent format.
 // Render is reusable across Renders, it holds configuration only.
-type Renderer struct{}
+type Renderer struct {
+	underlineHeadings bool
```

Comment:
```suggestion
	underlineHeadingsEnabled bool
```

Reply:
are you sure about that? it's too long for not much more info